lib/pages/common_widgets/goal_widget.dart

```diff
@@ -123,6 +181,56 @@ class GoalWidget extends StatelessWidget {
             );
   }
 
+  Future<dynamic> selectExtension(BuildContext context, Null doExtendGoal(), goalExtensionInput) {
+    return showDialog(
+      context: context,
+      builder: (BuildContext context) => AlertDialog(
+        title: const Text('Enter how many extra days you need:'),
+        content: //const Text('Selection will be here'),
+        
+        TextField(
+        controller: goalExtensionInput,
+        keyboardType: TextInputType.number,
+        decoration: InputDecoration(hintText: "Ex: 5"),
+        ),
+         
+        actions: <Widget>[
+        TextButton(
+          onPressed: () =>
+          Navigator.pop(context, 'Cancel'),
+          child: const Text('Cancel'),
+        ),
+        TextButton( 
+         onPressed: () {
+          if (goalExtensionInput.text != '') {
```

Comment:
is there a check somewhere to verify the extension isn't 0?

Reply:
done